packages/panzer/disc-fe/src/Panzer_IntegrationValues2_impl.hpp

```diff
@@ -0,0 +1,145 @@
+#ifndef PANZER_INTEGRATION_VALUES_2_IMPL_HPP
+#define PANZER_INTEGRATION_VALUES_2_IMPL_HPP
+
+/* Normally, we would fold this definition into the .cpp file since we
+   use ETI to compile this object. However, for unit testing the stand
+   alone device function swapQuadraturePoints, we need the definition
+   at the calling point to avoid requiring relocatable device code for
+   cuda. So for the IntegrationValues2 object, if there is a
+   standalone device member function, put it in this file (for unit
+   testing).
+ */
+namespace panzer {
+
+template<typename Scalar,
+         typename T0,typename T1,typename T2,typename T3,
+         typename T4,typename T5,typename T6,typename T7>
+void
+swapQuadraturePoints(int cell,
```

Comment:
Do you need to declare this as a kokkos inline function?

Reply:
It doesn't look like this is the main declaration, so maybe not.

Reply:
only in the declaration